src/versionbits.cpp

```diff
@@ -74,12 +79,16 @@ ThresholdState AbstractThresholdConditionChecker::GetStateFor(const CBlockIndex*
                 }
                 if (count >= nThreshold) {
                     stateNext = ThresholdState::LOCKED_IN;
+                } else if (height >= height_timeout) {
+                    stateNext = ThresholdState::FAILED;
```

Comment:
Do you have test coverage for threshold taking precedence over reaching timeout ? This diff doesn't break `feature_bip8.py`

```
@@ -77,10 +78,11 @@ ThresholdState AbstractThresholdConditionChecker::GetStateFor(const CBlockIndex*
                     }
                     pindexCount = pindexCount->pprev;
                 }
+                if (height >= height_timeout) {
+                    stateNext = ThresholdState::FAILED;
+                }
                 if (count >= nThreshold) {
                     stateNext = ThresholdState::LOCKED_IN;
-                } else if (height >= height_timeout) {
-                    stateNext = ThresholdState::FAILED;
                 }
                 break;
             }
```

Reply:
A unit test fails, although not because it explicitly tests this behavior. I've added such a test to feature_bip8.py.

However this particular diff wouldn't fail because it still moves to `LOCKED_IN` when the threshold is reached. You need to use `else if` in order for the `height >= height_timeout` check to take precedence.